libs-scala/scala-utils/src/test/scala/com/daml/scalautil/future/FutureConversionSpec.scala

```diff
@@ -0,0 +1,44 @@
+package com.daml.scalautil.future
+
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import com.daml.scalautil.future.FutureConversionSpec.TestException
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+
+import scala.concurrent.Future
+import scala.jdk.FutureConverters._
+
+class FutureConversionSpec extends AsyncWordSpec with Matchers {
+  import FutureConversion._
+  "converting a java CompletionStage into a scala Future" should {
+    "fail the future with the same exception as the CompletionStage" in {
+      val exception = new TestException
+      // build a completion stage that fails with CompletionException
+      // this is NOT the same as CompletableFuture.failedStage
+      val cs: CompletionStage[Unit] =
+        CompletableFuture.completedStage(()).thenApply(_ => throw exception)
```

Comment:
Perhaps write a "contract test" that ensures that this particular construction actually throws a `CompletionException`, in case a Java upgrade bites us somehow.

Reply:
Added assertions that using the regular converters, the future fails with CompletionException